chsdi/models/vector/bafu.py

```diff
@@ -2337,3 +2337,34 @@ class TranquillityGebiete(Base, Vector):
     the_geom = Column('the_geom', Geometry2D)
 
 register('ch.bafu.tranquillity-gebiete', TranquillityGebiete)
+
+
+class KlimaCO2AusstossGebaeude(Base, Vector):
+    __tablename__ = 'co2_ausstoss_gebaeude'
+    __table_args__ = ({'schema': 'klima', 'autoload': False})
+    __bodId__ = 'ch.bafu.klima-co2_ausstoss_gebaeude'
+    __template__ = 'templates/htmlpopup/bafu_klima_co2_ausstoss_gebaeude.mako'
+    __label__ = 'id'
+    id = Column('bgdi_id', Integer, primary_key=True)
```

Comment:
egid is the primary key here.
please convert egid in the database and the model to integer and set it as primary key:
```SQL
-- Firstly, remove bgdi_id primary key
ALTER TABLE klima.co2_ausstoss_gebaeude DROP CONSTRAINT pkey_co2_ausstoss_gebaeude;
-- convert egid to integer
alter table  klima.co2_ausstoss_gebaeude alter column egid type integer using egid::integer;
-- make egid PRIMARY KEY
ALTER TABLE klima.co2_ausstoss_gebaeude ADD PRIMARY KEY (egid);
```

After that you can remove the btree index on egid.

the same for the other new gwr building visualisation layer.

Reply:
This should be fixed.